common/processes_test.cpp

```diff
@@ -260,22 +272,238 @@ hard_sleep() {
 hard_sleep 10
 exit 0
 )delim";
+
 	auto ret = PrepareTestScript(script);
 	ASSERT_TRUE(ret);
 
-	procs::Process proc({TestScriptPath()});
-	auto err = proc.Start();
+	string ready_file = path::Join(path::DirName(TestScriptPath()), "test_script-ready");
+
+	auto proc = make_shared<procs::Process>(vector<string> {TestScriptPath()});
+	auto err = proc->Start();
 	ASSERT_EQ(err, error::NoError);
 
 	while (true) {
-		ifstream is(path::Join(path::DirName(TestScriptPath()), "test_script-ready"));
+		ifstream is(ready_file);
 		if (is.good()) {
 			break;
 		}
 	}
 
-	proc.Kill();
+	proc->Kill();
 
-	auto exit_status = proc.Wait();
+	auto exit_status = proc->Wait();
 	EXPECT_NE(exit_status, 0);
+
+	remove(ready_file.c_str());
+
+	auto start_time = chrono::steady_clock::now();
+
+	proc = make_shared<procs::Process>(vector<string> {TestScriptPath()});
+	err = proc->Start();
+	ASSERT_EQ(err, error::NoError);
+
+	while (true) {
+		ifstream is(ready_file);
+		if (is.good()) {
+			break;
+		}
+	}
+
+	auto &max_termination_time = ProcessesTestsHelper::GetMaxTerminationTime(*proc);
+	auto old_value = max_termination_time;
+	// Cut down a bit on the kill time in tests.
+	max_termination_time = chrono::seconds {1};
+
+	auto pid = ProcessesTestsHelper::GetNativeProc(*proc)->get_id();
+
+	// Kill by destruction.
+	proc.reset();
+
+	auto result = ::kill(pid, 0);
+	ASSERT_EQ(result, -1);
+	int errnum = errno;
+	// No such process.
+	ASSERT_EQ(errnum, ESRCH);
+
+	auto finish_time = chrono::steady_clock::now();
+	auto diff = finish_time - start_time;
+	EXPECT_LT(diff, chrono::seconds {10});
```

Comment:
Nit: Why not use the already hard-coded variable value?

Reply:
The fact that it's 10 is a coincidence. This value is not related to the `MAX_TERMINATION_TIME`, it's related to the `hard_sleep 10` which is in the shell script above.

Reply:
Ahh, I see. Fooled me!